GeometryCore/src/AABB.cpp

```diff
@@ -14,18 +14,36 @@ namespace geom_utils
         calculate(Polygon::makePolygon(tri));
     }
 
+    bool AABB2D::isValid(const AABB2D& aabb2d) 
+    {
+        if (aabb2d.pmax.x < aabb2d.pmin.x || aabb2d.pmax.y < aabb2d.pmin.y) return false;
+        else return true;
+    }
+
     bool AABB2D::contains(const FPoint2D& p) const
-    { 
-        return{}; 
+    {
+        if (p.x <= pmax.x && p.y <= pmax.y && p.x >= pmin.x && p.y >= pmin.y)  return true;
+        else return false;
     }
 
     void AABB2D::calculate(const Polygon& poly)
     {
-
+        if (!poly.empty() && poly.size() != 0) {
```

Comment:
Second check `poly.size() != 0` is redundant - if poly is not empty, then its size is surely larger than 0.